src/mlpack/tests/bayesian_linear_regression_test.cpp

```diff
@@ -0,0 +1,194 @@
+/**
+ * @file bayesian_linear_regression_test.cpp
+ * @author Clement Mercier
+ *
+ * Test for BayesianLinearRegression.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include <mlpack/core/data/load.hpp>
+#include <mlpack/methods/bayesian_linear_regression/bayesian_linear_regression.hpp>
+#include <mlpack/methods/linear_regression/linear_regression.hpp>
+
+#include <boost/test/unit_test.hpp>
+
+using namespace mlpack::regression;
+using namespace mlpack::data;
+
+BOOST_AUTO_TEST_SUITE(BayesianLinearRegressionTest);
```

Comment:
Not compulsory, but one more thing that can be manageable here is that we can migrate these tests from `boost` to `catch` in order continue this(#2523) great work started by @jeffin143. But once again this is not compulsory we can also manage this at later point of time, your choice as you are comfortable with.

Reply:
Yeah, definitely, we can add it to #2523 instead of doing it here if we like. :+1: